llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp

```diff
@@ -29,6 +29,38 @@
 
 using namespace llvm;
 
+static bool CC_AMDGPU_Custom_I1(unsigned ValNo, MVT ValVT, MVT LocVT,
+                                CCValAssign::LocInfo LocInfo,
+                                ISD::ArgFlagsTy ArgFlags, CCState &State) {
+  static bool IsWave64 =
+      State.getMachineFunction().getSubtarget<GCNSubtarget>().isWave64();
+
+  static const MCPhysReg SGPRArgsWave64[] = {
```

Comment:
We should maybe unroll the 64-bit case into unaligned 32-bit pieces, but it's going to complicate things. Probably best to leave that for a follow up patch later